src/apis/v1/popup/service.ts

```diff
@@ -0,0 +1,92 @@
+import { ErrorCodes, HttpException } from 'exceptions';
+import { PopupModel } from 'models';
+import { logger } from 'utils/logger';
+import { DocumentDto, UpdatePopupDto } from './dto/CreatePopupDto';
+
+export const createPopup = async (input: DocumentDto) => {
+  try {
+    const data = await PopupModel.create(input);
+    logger.info('Popup created successfully');
+
+    return data;
+  } catch (error) {
+    logger.error(`Error while creating popup: ${error}`);
+    throw new HttpException(
+      400,
+      error?.message || ErrorCodes.BAD_REQUEST.MESSAGE,
+      error?.code || ErrorCodes.BAD_REQUEST.CODE
+    );
+  }
+};
+
+export const deletePopup = async (id: string) => {
+  try {
+    const data = await PopupModel.findByIdAndDelete({
+      _id: id,
+    });
+    logger.info('Popup deleted successfully');
+    return data;
+  } catch (error) {
+    logger.error(`Error deleting popup: ${error} `);
+    throw new HttpException(400, ErrorCodes.BAD_REQUEST.MESSAGE, ErrorCodes.BAD_REQUEST.CODE);
+  }
+};
+
+export const revokedPopup = async (id: string) => {
+  try {
+    const data = await PopupModel.findByIdAndUpdate({ _id: id }, { is_revoked: true });
+
+    const result = await PopupModel.findById({ _id: id });
```

Comment:
Why do you find again? is that necessary?

Reply:
update and return new value is updated after update

Reply:
I think it is unnecessary, because when you update you only care process is success or not, and then when success you can call refresh()